cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaAgentClient.java

```diff
@@ -0,0 +1,117 @@
+            KeyStore trustStore = KeyStore.getInstance("PKCS12");
+            trustStore.load(new ByteArrayInputStream(
+                Util.decodeFromSecret(clusterCaCertSecret, "ca.p12")),
+                new String(Util.decodeFromSecret(clusterCaCertSecret, "ca.password"), StandardCharsets.UTF_8).toCharArray());
+
+            String trustManagerFactoryAlgorithm = TrustManagerFactory.getDefaultAlgorithm();
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(trustManagerFactoryAlgorithm);
+            trustManagerFactory.init(trustStore);
+            char[] keyPassword = new String(Util.decodeFromSecret(coKeySecret, "cluster-operator.password"), StandardCharsets.UTF_8).toCharArray();
+            KeyStore coKeyStore = KeyStore.getInstance("PKCS12");
+            coKeyStore.load(new ByteArrayInputStream(
+                Util.decodeFromSecret(coKeySecret, "cluster-operator.p12")),
+                keyPassword
+            );
+
+            String keyManagerFactoryAlgorithm = KeyManagerFactory.getDefaultAlgorithm();
+            KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(keyManagerFactoryAlgorithm);
+            keyManagerFactory.init(coKeyStore, keyPassword);
+
+            sslContext.init(keyManagerFactory.getKeyManagers(), trustManagerFactory.getTrustManagers(), null);
+
+
+            HttpRequest req = HttpRequest.newBuilder()
+                    .uri(uri)
+                    .GET()
+                    .build();
+
+            var response =  HttpClient.newBuilder()
+                    .sslContext(sslContext)
+                    .build()
+                    .send(req, HttpResponse.BodyHandlers.ofString());
+            if (response.statusCode() != 200) {
+                throw new RuntimeException("Unexpected HTTP status code: " + response.statusCode());
+            }
+            return response.body();
+        } catch (GeneralSecurityException | IOException | InterruptedException e) {
+            throw new RuntimeException("Failed to send HTTP request to Kafka Agent", e);
+        }
+    }
+
+    /**
+     * Gets broker state by sending HTTP request to the /v1/broker-state endpoint of the KafkaAgent
+     *
+     * @param podName Name of the pod to interact with
+     * @return A BrokerState that contains broker state and recovery progress.
+     *         -1 is returned for broker state if the http request failed or returned non 200 response.
+     *         Null value is returned for recovery progress if broker state is not 2 (RECOVERY).
+     */
+    public BrokerState getBrokerState(String podName) {
+        BrokerState brokerstate = new BrokerState(-1, null);
+        String host = DnsNameGenerator.podDnsName(namespace, KafkaResources.brokersServiceName(cluster), podName);
+        try {
+            URI uri = new URI("https://" +  host + ":" + BROKER_STATE_HTTPS_PORT + BROKER_STATE_REST_PATH);
```

Comment:
You should use the `new URI("https", null, host, BROKER_STATE_HTTPS_PORT, BROKER_STATE_REST_PATH, null, null)` constructor  (which will avoid the URI constructor having to parse a single URI into its constituent parts).